scipy/linalg/_basic.py

```diff
@@ -243,6 +276,20 @@ def solve(a, b, lower=False, overwrite_a=False,
                                  overwrite_b=overwrite_b)
         _solve_check(n, info)
         rcond, info = sycon(lu, ipvt, anorm)
+    # Diagonal case
+    elif assume_a == 'dia':
+        x = (b1.T / np.diag(a1)).T
+    # Tri-diagonal case
+    elif assume_a == 'tri':
+        a1 = a1.T if transposed else a1
+        dl, d, du = np.diag(a1, -1), np.diag(a1, 0), np.diag(a1, 1)
+        _gtsv = get_lapack_funcs('gtsv', (a1, b1))
+        x, info = _gtsv(dl, d, du, b1, False, False, False, overwrite_b)[3:]
+    # Triangular case
+    elif assume_a in {'ltr', 'utr'}:
+        lower = assume_a == 'ltr'
+        x = solve_triangular(a1, b1, lower=lower, overwrite_b=overwrite_b,
```

Comment:
This is doubling the performance loss. We already checked everything hence you can call `trtrs` directly after `trcon`.

Reply:
Yeah, I know; I just did this for simplicity to get something in place, and I figured you'd know exactly what needs to be done ; )